client/block-builder/src/lib.rs

```diff
@@ -203,7 +204,13 @@ where
 	/// Push onto the block's list of extrinsics.
 	///
 	/// This will ensure the extrinsic can be validly executed (by executing it).
-	pub fn push(&mut self, xt: <Block as BlockT>::Extrinsic) -> Result<(), Error> {
+	/// If `ensure_block_limit` is provided, this function will ensure that the execution
+	/// of the block will not exceed the provided limit.
+	pub fn push(
+		&mut self,
+		xt: <Block as BlockT>::Extrinsic,
+		ensure_block_limit: Option<(bool, usize)>,
```

Comment:
```suggestion
		ensure_block_limit: Option<Limit>,
```

using some struct here would be more expressive.